lib/mayaUsd/render/mayaToHydra/renderOverride.cpp

```diff
@@ -830,6 +830,15 @@ void MtohRenderOverride::ClearHydraResources()
     _delegates.clear();
     _defaultLightDelegate.reset();
 
+    // Cleanup internal context data that keep references to data that is now
+    // invalid.
+#if PXR_VERSION >= 2108
+    _engine.ClearTaskContextData();
+#else
+    for (const auto& token : HdxTokens->allTokens)
```

Comment:
Pre-21.08, there was no clear function and there is no way to iterate on the context data... so I just clear some of the potentially cached tokens, at least the ones that were causing this particular crash, related to the light and shadow system.